### PR TITLE
POC: UI e2e testing (Cypress)

### PR DESCRIPTION
**Pull request by jonycfu**
** PR checklist **

- [ X] Ensure you have added or ran the appropriate tests for your PR.
- [ X] DCO signed

**What type of PR is this?**
Feature
**What this PR does / why we need it**:
E2E UI Testing (smoke-tests : no blank page breakage)
**Which issue(s) this PR fixes**:
SC 27301
**Special notes for your reviewer**:
Local Working (below); WIP testing on PR event on gh-actions
https://www.loom.com/share/7f576a39d2fd45ff91a06929b3ba4811
**Does this PR introduce a user-facing change?**:
No

NONE

### CODE REVIEW

**Comment by jonycfu** [comment]
The large file sizes are due to the fact that we are committing the raw json files. A further improvement will be made to make it more dynamic in the future, reusing cli to generate said `json` in the project

**Comment by jonycfu** [comment]
After much thought, I will improve this implementation to generate jsons from piperider cli directly. 
Current state will also include jsons into build (not good)

**Comment by jonycfu** [comment]
> After much thought, I will improve this implementation to generate jsons from piperider cli directly. Current state will also include jsons into build (not good)

Update: found out that this may require `expect` script to handle the inputs to `init` and `compare-reports` commands. 
Not sure how complicated it is to implement or if it's possible. @ctiml is this possible?

As for the "included jsons into the build" mentioned earlier, I can remove them from `public/` instead. So no issues there. 

The main question is whether we want to version control data json files specifically for UI e2e purposes vs. having the actual profiler cli generate the data to consume in the e2e report tests, which make this a full integration test. Either way, there's a tradeoff between mock data maintenance and scope size (pure FE only + more mocks vs. fully integration test + less mocks)

**Review by neighborhood999 — CHANGES_REQUESTED** [review]

File: static_report/package.json [hunk 1]
```diff
@@ -44,14 +45,18 @@
   },
   "scripts": {
     "setup": "run-s typing:* embed:html",
-    "start:single": "npm run setup && PORT=3000 REACT_APP_SINGLE_REPORT=true react-scripts start",
-    "start:comparison": "npm run setup && PORT=3001 REACT_APP_SINGLE_REPORT=false react-scripts start",
+    "serve:single": "PORT=3000 REACT_APP_SINGLE_REPORT=true react-scripts start",
```
[the remark this answers is not in the document]
Comment (jonycfu) [reply static_report/package.json:49]: This actually doesn't change. we will still use 
```
    "start:single": "run-s setup serve:single",
    "start:comparison": "run-s setup serve:comparison",
```

[the remark this answers is not in the document]
Comment (jonycfu) [reply static_report/package.json:49]: image

**Comment by jonycfu** [comment]
@neighborhood999 @ctiml 
I need a final review from you guys. Build now works. 
Have updated readme section documenting the E2E UI process for Cypress. 

Thanks in advance 🙏

File: static_report/package.json [hunk 2]
```diff
@@ -15,6 +15,7 @@
     "d3": "^7.4.4",
     "date-fns": "^2.28.0",
     "framer-motion": "^6.3.4",
+    "http-server": "^14.1.1",
```
Comment (neighborhood999) [inline static_report/package.json:18]: Move to  `devDependencies`?

File: static_report/package.json [hunk 3]
```diff
@@ -32,6 +33,7 @@
     "@types/react": "^18.0.14",
     "@types/react-dom": "^18.0.5",
     "chalk": "^5.0.1",
+    "cypress": "^10.3.0",
```
Comment (neighborhood999) [inline static_report/package.json:36]: Same as the above question.

**Review by neighborhood999 — APPROVED** [review]

**Pull request merged**